frontend/src/components/dashboard/Analysis.tsx

```diff
@@ -146,15 +193,34 @@ const Analysis = ({
   const tooltipRef = useRef<HTMLDivElement>(null);
   const tooltipHovered = useRef(false);
 
-  const xAxisConfig = useMemo(() => getAxisConfig(xAxis), [xAxis]);
-  const yAxisConfig = useMemo(() => getAxisConfig(yAxis), [yAxis]);
+  const setXAxis = useCallback(
+    (newAxis: AxesCriterionType) =>
+      setAxis(dispatch, state.xAxis, state.yAxis, newAxis, "x", state.splits),
+    [dispatch, state.xAxis, state.yAxis, state.splits],
+  );
+
+  const setYAxis = useCallback(
+    (newAxis: AxesCriterionType) =>
+      setAxis(dispatch, state.yAxis, state.xAxis, newAxis, "y", state.splits),
+    [dispatch, state.xAxis, state.yAxis, state.splits],
+  );
```

Comment:
with the suggestion above, this would be massively simplified as there would be no dependency on `state`.